### PR TITLE
AKS: Add ephemeral disk volume type and nvme perf tier support to Azure Container Storage

### PR DESCRIPTION
**Pull request by mukhoakash**
This PR adds support to change the ephemeral disk volume type ephemeral nvme perf tier for Azure Container Storage installations on AKS cluster.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -13,12 +13,18 @@ Pending
 +++++++
 * Vendor new SDK and bump API version to 2024-04-02-preview.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:14]: Please include this note as part of 5.0.0b2 as well

**Review by FumingZhang — COMMENTED** [review]
Could you please add some live test cases for the newly added options?

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by FumingZhang** [comment]
I've fixed the common style/lint issue in aks-preview via another PR #7741, please rebase from main.

**Comment by mukhoakash** [comment]
Triggered [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=96749565&view=results): 
- test_aks_create_with_azurecontainerstorage
- test_aks_update_with_azurecontainerstorage
- test_aks_create_with_azurecontainerstorage_with_nodepool_name
-

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -11164,6 +11219,60 @@ def test_aks_update_with_azurecontainerstorage(self, resource_group, resource_gr
...
+        ])
+
+        # enabling or disabling azurecontainerstorage will not affect any field in the cluster.
+        # the only check we should perform is to verify that the cluster is provisioned successfully.
+
+        # update: enable-azure-container-storage
+        update_cmd = 'aks update --resource-group={resource_group} --name={name} --yes --output=json ' \
+                     '--enable-azure-container-storage ephemeralDisk --storage-pool-option NVMe ' \
+                     '--ephemeral-disk-volume-type PersistentVolumeWithAnnotation ' \
+                     '--ephemeral-disk-nvme-per-tier Standard'
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:11252]: > ERROR    cli.azure.cli.core.azclierror:azlogging.py:212 unrecognized arguments: --ephemeral-disk-nvme-per-tier Standard
```suggestion
                     '--ephemeral-disk-nvme-perf-tier Standard'
```

Comment (mukhoakash) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py]: Fixed

**Comment by mukhoakash** [comment]
Triggered [new live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=96772763&view=results):

test_aks_create_with_azurecontainerstorage
test_aks_update_with_azurecontainerstorage
test_aks_create_with_azurecontainerstorage_with_nodepool_name
test_aks_create_with_azurecontainerstorage_with_ephemeral_disk_parameters

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**